lib/iris/pandas.py

```diff
@@ -518,88 +518,298 @@ def _get_base(array):
+    outlist = []
+    ndims = cube.ndim
+    for dimn in range(ndims):
+        for coord in cube.coords(dimensions=dimn, dim_coords=True):
+            if not coord:
+                outlist += [dimn, None]
+            else:
+                outlist += [dimn, coord]
+    return outlist
+
+
 def as_series(cube, copy=True):
     """
     Convert a 1D cube to a Pandas Series.
 
-    Args:
+    .. deprecated:: 3.3.2
 
-        * cube - The cube to convert to a Pandas Series.
+        This function is scheduled for removal in a future release, being
+        replaced by :func:`iris.pandas.as_data_frame`, which offers improved
+        multi dimension handling.
 
+    Args:
+        * cube - The cube to convert to a Pandas Series.
     Kwargs:
-
         * copy - Whether to make a copy of the data.
                  Defaults to True. Must be True for masked data.
-
     .. note::
-
         This function will copy your data by default.
         If you have a large array that cannot be copied,
         make sure it is not masked and use copy=False.
-
     """
+    message = (
+        "iris.pandas.as_series has been deprecated, and will be removed in a "
+        "future release. Please use iris.pandas.as_data_frame instead."
+    )
+    warn_deprecated(message)
+
     data = cube.data
     if ma.isMaskedArray(data):
         if not copy:
             raise ValueError("Masked arrays must always be copied.")
         data = data.astype("f").filled(np.nan)
     elif copy:
         data = data.copy()
-
     index = None
     if cube.dim_coords:
         index = _as_pandas_coord(cube.dim_coords[0])
-
     series = pandas.Series(data, index)
     if not copy:
         _assert_shared(data, series)
-
     return series
 
 
-def as_data_frame(cube, copy=True):
+def as_data_frame(
+    cube,
+    copy=True,
+    add_aux_coord=False,
+    add_global_attributes=None,
+):
     """
-    Convert a 2D cube to a Pandas DataFrame.
-
-    Args:
+    :attr:`~iris.cube.Cube.dim_coords` and :attr:`~iris.cube.Cube.data` are flattened into a long-style
+    :class:`~pandas.DataFrame`.  Other :attr:`~iris.cube.Cube.aux_coords` and :attr:`~iris.cube.Cube.attributes`
+    may be optionally added as additional :class:`~pandas.DataFrame` columns.
 
-        * cube - The cube to convert to a Pandas DataFrame.
-
-    Kwargs:
-
-        * copy - Whether to make a copy of the data.
-                 Defaults to True. Must be True for masked data
-                 and some data types (see notes below).
+    Parameters
+    ----------
+    :class:`~iris.cube.Cube`
+        The :class:`Cube` to be converted to a Pandas `DataFrame`.
+    copy : bool, default=True
+        Whether the Pandas `DataFrame` is a copy of the the Cube :attr:`~iris.cube.Cube.data`.
+        This option is provided to help with memory size concerns.
+    add_aux_coord : bool, default=False
+        If True, add all :attr:`~iris.cube.Cube.aux_coords` to add to the returned `DataFrame`.
```

Comment:
Mainly for speed of execution.  But happy to default to `True` with an added note about trying `False` if conversion seems slow.

Reply:
Ah, good point, I hadn't considered performance. Happy to leave as-is.

If you action https://github.com/SciTools/iris/pull/4669#discussion_r1003373284, it would be neatest to have everything covered by a single parameter.